### PR TITLE
Separate tests into different script files

### PR DESCRIPTION
**Pull request by jspayd**
To help migrate to Buildbot. The test suite can be run with `make test`.

Notes:
1. You may need to install dependencies first to test locally, all of which are listed in `.travis.yml`. I'd recommend installing the Python packages in a virtualenv.
```
sudo apt-get update
sudo apt-get install -y nodejs
sudo apt-get install -y python-demjson
pip install -r requirements.txt
pip install demjson
```
2. The `tests/test_edxapp_play.sh` test requires the calling user to have passwordless sudo (or can be run with sudo).
3. The tests only need to work on Linux, but based on my testing, they seem to work on macOS too.

@stvstnfrd @caesar2164
CC: @potsui @jlikhuva

### CODE REVIEW

File: tests/test_jinja_syntax.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+cd playbooks
```
Comment (stvstnfrd) [inline tests/test_jinja_syntax.sh]: does this part actually matter? What happens if we run from the project root?

File: tests/test_jinja_syntax.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+for j2 in $(git diff --name-only refs/heads/master..$(git rev-parse --abbrev-ref HEAD) |grep -E '.+\.j2$|.+\.yml'); do
+  python ../tests/jinja_check.py ../$j2
```
Comment (stvstnfrd) [inline tests/test_jinja_syntax.sh]: This seems odd.. see note above about trying to run from project root.

File: tests/test_edxapp_play.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+ROOT_DIR=${PWD}
+cd playbooks/edx-east
+output_dir="${PWD}/test_output/env-dep"
+mkdir -p ${output_dir}
+ansible-playbook -i localhost, -c local --tags edxapp_cfg edxapp.yml -e edxapp_user=`whoami` -e edxapp_app_dir=${output_dir} -e edxapp_code_dir=${output_dir} -e EDXAPP_CFG_DIR=${output_dir}
+
+root_dir=${output_dir}
+environment_deployments="."
```
Comment (stvstnfrd) [inline tests/test_edxapp_play.sh]: Move these two lines (8,9) into the `validate_templates` file,
then execute that file via the `Makefile`.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+	| sort \
+	| awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: test
+test:  ## Tests everything
+	tests/test_yaml_syntax.sh
+	tests/test_json_syntax.sh
+	tests/test_jinja_syntax.sh
+	tests/test_role_syntax.sh
+	tests/test_edxapp_play.sh
```
Comment (stvstnfrd) [inline Makefile]: Add `validate_template` here.

File: tests/test_edxapp_play.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (stvstnfrd) [inline tests/test_edxapp_play.sh]: Rename all files to remove `test_` prefix and `.sh` suffix.

File: tests/test_jinja_syntax.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+for j2 in $(git diff --name-only refs/heads/master..$(git rev-parse --abbrev-ref HEAD) |grep -E '.+\.j2$|.+\.yml'); do
```
Comment (stvstnfrd) [inline tests/test_jinja_syntax.sh]: Decide whether to split this out into a variable.

**Comment by jspayd** [comment]
@stvstnfrd  I just realized (after pushing) that without any updates to `validate_templates.sh`, `FAIL` is never initialized to `0`, so this error happens.
```
/Users/vptluser/configuration-workbench/configuration/tests/validate_templates.sh: line 34: [: : integer expression expected
```
Should I go ahead and add that into the script even though edX's version doesn't have it?

**Comment by jspayd** [comment]
@stvstnfrd Looks like Travis is now saying
```
tests/test_edx_east_roles.sh: 23: tests/test_edx_east_roles.sh: source: not found
```
because `test_edx_east_roles.sh` uses `source` and not `.`. Should I change that script?

**Comment by jspayd** [comment]
@stvstnfrd Are there more changes I should make or does this all look good?

File: tests/validate_templates.sh [hunk 7]
```diff
@@ -3,6 +3,7 @@
 # The 'root_dir' and 'environment_deployments' variables
 # should be set when we source this.
 
+FAIL=0
```
Comment (stvstnfrd) [inline tests/validate_templates.sh]: Could we move this to `tests/test_edx_east_roles.sh` to keep this file unforked for now?

**Pull request merged**